File: 프로그래머스/0/120851. 숨어있는 숫자의 덧셈 （1）/숨어있는 숫자의 덧셈 （1）.py
def solution(my_string):
    answer = 0  # 숫자의 합을 저장할 변수

    for m in my_string:  # 문자열의 각 문자를 순회
        if m.isnumeric():  # 숫자인 경우에만
            answer += int(m)  # 정수로 변환하여 더한다.

    return answer  # 최종 결과 반환

# 테스트 예제 실행
print(solution("aAb1B2cC34oOp")) # 10
print(solution("1a2b3c4d123"))   # 16

File: 프로그래머스/0/120851. 숨어있는 숫자의 덧셈 （1）/숨어있는 숫자의 덧셈 （1）.java
public class Solution {
    public static int solution(String my_string) {
        int answer = 0;

        for (char m : my_string.toCharArray()) {
            int code = (int)m;
            if (code >= 48 && code <= 57) {
                answer += (code - 48);
            }
        }

        return answer;
    }

    public static void main(String[] args) {
        System.out.println(solution("aAb1B2cC34oOp"));
    }
}